Source/Urho3D/CSharp/Managed/Math/MathDefs.cs

```diff
@@ -546,5 +546,54 @@ public static uint FloatToRawIntBits(float value)
                 return *(uint*)&value;
             }
         }
+
+        /// Return a random float between 0.0 (inclusive) and 1.0 (exclusive).
+        public static float Random() { return Rand() / 32768.0f; }
+
+        /// Return a random float between 0.0 and range, inclusive from both ends.
+        public static float Random(float range) { return Rand() * range / 32767.0f; }
+
+        /// Return a random float between min and max, inclusive from both ends.
+        public static float Random(float min, float max) { return Rand() * (max - min) / 32767.0f + min; }
+
+        /// Return a random integer between 0 and range - 1.
+        /// @alias{RandomInt}
+        public static int Random(int range) { return (int)(Random() * range); }
+
+        /// Return a random integer between min and max - 1.
+        /// @alias{RandomInt}
+        public static int Random(int min, int max) { var range = (float)(max - min); return (int)(Random() * range) + min; }
+
+        /// Return a random normal distributed number with the given mean value and variance.
+        public static float RandomNormal(float meanValue, float variance) { return RandStandardNormal() * (float)Math.Sqrt(variance) + meanValue; }
+
+
+        private static uint randomSeed = 1;
```

Comment:
I am not quite sure what is the point of these functions since we can use `System.Random`. Especially since C# and C++ sides keep separate seeds so any chance for matching results is ruined right there.

Reply:
I just brought it for consistency. Also there is a standard deviation approximation which is missing from System.Random. I can pass Random as an argument but that is weird.

Reply:
Maybe you just adopt RandomEngine for C# and drop old global functions?
https://github.com/rokups/rbfx/blob/master/Source/Urho3D/Math/RandomEngine.h